hellodagger/src/main/java/com/airbnb/mvrx/helloDagger/HelloFragment.kt

```diff
@@ -0,0 +1,41 @@
+package com.airbnb.mvrx.helloDagger
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.airbnb.mvrx.*
+import kotlinx.android.synthetic.main.fragment_hello.*
+
+class HelloFragment: BaseMvRxFragment() {
+
+    private val viewModel: HelloViewModel by fragmentViewModel()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_hello, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        helloButton.setOnClickListener { viewModel.sayHello() }
+    }
+
+    override fun invalidate() {
+        withState(viewModel) { state: HelloState ->
+            when (state.message) {
+                is Uninitialized, is Loading -> {
+                    messageTextView.text = getString(R.string.helloFragmentLoadingText)
+                    helloButton.isEnabled = false
+                }
+                is Success -> {
+                    messageTextView.text = state.message()
+                    helloButton.isEnabled = true
+                }
+                is Fail -> {
+                    messageTextView.text = getString(R.string.helloFragmentFailureText)
+                    helloButton.isEnabled = true
+                }
+            }
+        }
+    }
```

Comment:
This can be simplified as
```
override fun invalidate() = withState(viewModel) { state ->
    helloButton.isEnabled = state.message !is Loading
    messageView.text = when (state.message) {
        is Uninitialized, is Loading ->  getString(R.string.helloFragmentLoadingText)
        is Success -> state.message()
        is Fail ->  getString(R.string.helloFragmentFailureText)
    }
}
```

Reply:
Fixed, thanks for the suggestion.